From 23941f3a9734877af37d4ce2b5f4154588a8a356 Mon Sep 17 00:00:00 2001
From: ngc7331 <ngc7331@outlook.com>
Date: Fri, 6 Dec 2024 08:53:46 +0800
Subject: [PATCH] feat: remove cert mismatch check

---
 cmd/derper/cert.go | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/cmd/derper/cert.go b/cmd/derper/cert.go
index 623fa376f..187c78799 100644
--- a/cmd/derper/cert.go
+++ b/cmd/derper/cert.go
@@ -94,9 +94,9 @@ func (m *manualCertManager) TLSConfig() *tls.Config {
 }
 
 func (m *manualCertManager) getCertificate(hi *tls.ClientHelloInfo) (*tls.Certificate, error) {
-	if hi.ServerName != m.hostname && !m.noHostname {
-		return nil, fmt.Errorf("cert mismatch with hostname: %q", hi.ServerName)
-	}
+	// if hi.ServerName != m.hostname && !m.noHostname {
+	// 	return nil, fmt.Errorf("cert mismatch with hostname: %q", hi.ServerName)
+	// }
 
 	// Return a shallow copy of the cert so the caller can append to its
 	// Certificate field.
-- 
2.43.0

